doc/plugin_tutorial/tuto0/src/g_tuto0.mlg

```diff
@@ -7,12 +7,59 @@ open Ltac_plugin
 
 }
 
+(*** Printing messages ***)
+
+(*
+ * This defines a command that prints HelloWorld.
+ * Note that Feedback.msg_notice can be used to print messages.
+ *)
 VERNAC COMMAND EXTEND HelloWorld CLASSIFIED AS QUERY
 | [ "HelloWorld" ] -> { Feedback.msg_notice (strbrk Tuto0_main.message) }
 END
 
+(*
+ * This is a tactic version of the same thing.
+ *)
 TACTIC EXTEND hello_world_tactic
 | [ "hello_world" ] ->
   { let _ = Feedback.msg_notice (str Tuto0_main.message) in
     Tacticals.New.tclIDTAC }
 END
+
+(*** Printing warnings ***)
+
+(*
+ * This defines a command that prints HelloWorld as a warning.
+ * Note that Feedback.msg_warning can be used to print warnings.
+ *)
+VERNAC COMMAND EXTEND HelloWarning CLASSIFIED AS QUERY
+| [ "HelloWarning" ] -> { Feedback.msg_warning (strbrk Tuto0_main.message) }
```

Comment:
msg_warning shouldn't be used directly, instead you're supposed to define a warning through `CWarnings.create` as a global letin and call the resulting function. See eg https://github.com/coq/coq/blob/589aaf4f97d5cfcdabfda285739228f5ee52261f/library/goptions.ml#L225-L228 (random cwarnings.create from the source)

Reply:
done